### PR TITLE
Rename frame to advance, and start from advance 0 (closes #108)

### LINKED ISSUE #108: Frame: 1 -> Advance: 0
It would be useful to be able to use an initial frame of 0.
This way, once the delay is set up correctly, you can verify for tests purposes that you are on the correct frame, as the correct spread will appear at the first frame.

### PR DESCRIPTION
**Pull request by zaksabeast**
This pull request renames `frame` to `advance` for the end user, and shows RNG results starting from `advance 0`.

Internally, `frame` was being used to represent both the number of RNG advances that had past, as well as the Pokemon a game produces at that point.  Along with the rename for users, the code and files have been renamed to represent these different concepts and match [the recent RaidFinder change](https://github.com/Admiral-Fish/RaidFinder/commit/04611fb8bd2f93621e7e84bdb4040f25ab82e356).

### CODE REVIEW

**Comment by zaksabeast** [comment]
> For ui stuff, should it be Advance or Advances.

Is there a place this was missed?

> All uses of Starting Advance should probably be changed to Initial Advances or something. Starting Advance sounds weird to me.

Sounds good.  I wasn't sure which you wanted to go with and figured you'd provide a direction here.

> Do the outputs of generator and searchers match in Gen 4/5 in terms of advances.

Yes, I've been doing comparisons between this branch and the master branch for Gen 3, 4, 5, and Researcher to make sure I didn't break anything.  That said, it's entirely possible I missed something, so I'd appreciate you double checking (case in point, I just checked again and Dream Radar needed to be modified).

The one thing I forgot to mention when making the PR is Gen 3 Eggs seem to be off between the master branch and this branch. 
 Adding back the ` - 1` when setting the initial advances restores everything (which makes sense since that's the only change that isn't a rename).  I'm not familiar with Gen 3 eggs, so I wanted to get your thoughts on this.

The other requested changes have been made.

**Comment by Admiral-Fish** [comment]
I remember seeing it in the model headers where it was `Advance`.

I'll slowly start looking at matching outputs.

File: Forms/Gen5/Stationary5.cpp [hunk 1]
```diff
@@ -179,9 +179,9 @@ void Stationary5::generate()
     }
 
     u8 genderRatio = ui->comboBoxGeneratorGenderRatio->currentData().toInt();
-    Generator5 generator(maxResults, startingFrame, seed, tid, sid, offset,
+    Generator5 generator(maxResults, startingAdvance, seed, tid, sid, offset,
         static_cast<Method>(ui->comboBoxGeneratorMethod->currentData().toInt()));
-    FrameCompare compare(ui->comboBoxGeneratorAbility->currentIndex(), ui->comboBoxGeneratorAbility->currentIndex(),
+    AdvanceCompare compare(ui->comboBoxGeneratorAbility->currentIndex(), ui->comboBoxGeneratorAbility->currentIndex(),
```
[the remark this answers is not in the document]
Comment (zaksabeast) [reply Forms/Gen5/Stationary5.cpp:184]: Are there places that were missed?

File: Forms/Gen5/Event5.hpp [hunk 2]
```diff
@@ -26,9 +26,9 @@
 
 class EventGeneratorModel5;
 class EventSearcherModel5;
-class Frame;
-template <typename FrameType>
-class SearcherFrame5;
+class State;
+template <typename AdvanceType>
```
[the remark this answers is not in the document]
Comment (zaksabeast) [reply Forms/Gen5/Event5.hpp:30]: Ah geez - thanks for catching this.

File: Core/Gen5/Generators/DreamRadarGenerator.cpp [hunk 3]
```diff
@@ -49,56 +49,56 @@ DreamRadarGenerator::DreamRadarGenerator(u32 initialFrame, u32 maxResults, u16 t
...
     {
         rng.next();
     }
 
     MT mt(seed >> 32);
-    mt.advanceFrames(9); // Initial advances
-    mt.advanceFrames((initialFrame - 1) * 2); // Starting frame
-    mt.advanceFrames(ivAdvances); // Slot advances
+    mt.advance(9); // Initial advances
+    mt.advance((initialAdvances)*2); // Starting advance
```
[the remark this answers is not in the document]
Comment (zaksabeast) [reply Core/Gen5/Generators/DreamRadarGenerator.cpp:67]: Good catch - I mass searched for `initialFrame - 1`, and the formatter seems to have left the parenthesis.

File: Core/Gen5/Generators/DreamRadarGenerator.cpp [hunk 4]
```diff
@@ -23,9 +23,9 @@
 #include <Core/RNG/RNGList.hpp>
 #include <Core/Util/Utilities.hpp>
 
-DreamRadarGenerator::DreamRadarGenerator(u32 initialFrame, u32 maxResults, u16 tid, u16 sid, u8 genderRatio, Method method,
-                                         const FrameFilter &filter, const QVector<DreamRadarSlot> &radarSlots) :
-    Generator(initialFrame, maxResults, tid, sid, genderRatio, method, filter),
+DreamRadarGenerator::DreamRadarGenerator(u32 initialAdvances, u32 maxAdvances, u16 tid, u16 sid, u8 genderRatio, Method method,
+                                         const StateFilter &filter, const QVector<DreamRadarSlot> &radarSlots) :
+    Generator(initialAdvances, maxAdvances, tid, sid, genderRatio, method, filter),
     pidAdvances(0),
```
[the remark this answers is not in the document]
Comment (zaksabeast) [reply Core/Gen5/Generators/DreamRadarGenerator.cpp:29]: Aside from comparing results with the master branch, is there anything else that should be done to verify this works as expected?

File: Core/Gen5/Generators/EventGenerator5.cpp [hunk 5]
```diff
@@ -22,50 +22,49 @@
...
+EventGenerator5::EventGenerator5(u32 initialAdvances, u32 maxAdvances, u16 tid, u16 sid, u8 genderRatio, Method method,
+                                 const StateFilter &filter, const PGF &parameters) :
+    Generator(initialAdvances, maxAdvances, tid, sid, genderRatio, method, filter), parameters(parameters)
 {
     u16 cardXOR = parameters.getTID() ^ parameters.getSID();
     xorValue = parameters.isEgg() ? (tid ^ sid) : cardXOR;
     tsv = xorValue >> 3;
 
-    advances = parameters.getAdvances();
+    advanceParameter = parameters.getAdvances();
```
[the remark this answers is not in the document]
Comment (zaksabeast) [reply Core/Gen5/Generators/EventGenerator5.cpp:33]: Yep, will do

File: Core/Gen5/Generators/DreamRadarGenerator.cpp (hunk 4 above)
[the remark this answers is not in the document]
Comment (Admiral-Fish) [reply Core/Gen5/Generators/DreamRadarGenerator.cpp:29]: If it matches master then it's probably fine. The only other thing would be testing ingame

File: Core/Gen3/Generators/GameCubeGenerator.cpp [hunk 6]
```diff
@@ -200,10 +200,10 @@ QVector<GameCubeFrame> GameCubeGenerator::generateColoShadow(u32 seed) const
         u32 pid;
         for (auto lock = locks.rbegin(); lock != locks.rend(); lock++)
         {
-            // Temporary PID: 2 frames
-            // IVs: 2 frames
-            // Ability: 1 frame
-            go.advanceFrames(4);
+            // Temporary PID: 2 advances
+            // IVs: 2 advances
+            // Ability: 1 state
```
[the remark this answers is not in the document]
Comment (zaksabeast) [reply Core/Gen3/Generators/GameCubeGenerator.cpp:205]: It looks like this was changed to `advance`.  Is there something missing?

File: Core/Gen3/Generators/GameCubeGenerator.cpp [hunk 7]
```diff
@@ -71,48 +71,48 @@ QVector<GameCubeFrame> GameCubeGenerator::generateXDColo(u32 seed) const
...
 
         for (auto lock = locks.rbegin(); lock != locks.rend(); lock++)
         {
-            // Temporary PID: 2 frames
-            // IVs: 2 frames
-            // Ability: 1 frame
-            go.advanceFrames(5);
+            // Temporary PID: 2 advances
+            // IVs: 2 advances
+            // Ability: 1 state
```
[the remark this answers is not in the document]
Comment (zaksabeast) [reply Core/Gen3/Generators/GameCubeGenerator.cpp:114]: It looks like this was changed to `advance`.  Is there something missing?

**Pull request merged**